File: src/main/java/ca/dijital/hydro/Reading.java
package ca.dijital.hydro;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Date;

public class Reading {

	private long rid;
	private String id;
	private Date readTime;
	private double level;
	private double discharge;
	private String province;
	
	public static Reading fromResultSet(ResultSet rslt) throws SQLException {
		Reading r = new Reading();
		r.setRid(rslt.getLong("rid"));
		r.setDischarge(rslt.getDouble("discharge"));
		r.setId(rslt.getString("id"));
		r.setLevel(rslt.getDouble("level"));
		r.setProvince(rslt.getString("prov"));
		r.setReadTime(rslt.getTimestamp("readtime"));
		return r;
	}
	
	public String toString() {
		return String.format("[Reading: %s, %s, %f, %f]", getId(), getReadTime().toString(), getLevel(), getDischarge());
	}
	
	public long getRid() {
		return rid;
	}
	public void setRid(long rid) {
		this.rid = rid;
	}
	public String getId() {
		return id;
	}
	public void setId(String id) {
		this.id = id;
	}
	public Date getReadTime() {
		return readTime;
	}
	public void setReadTime(Date readingTime) {
		this.readTime = readingTime;
	}
	public double getLevel() {
		return level;
	}
	public void setLevel(double level) {
		this.level = level;
	}
	public double getDischarge() {
		return discharge;
	}
	public void setDischarge(double discharge) {
		this.discharge = discharge;
	}
	public String getProvince() {
		return province;
	}
	public void setProvince(String province) {
		this.province = province;
	}
	
}

File: svc/reading.js
class Reading {

	constructor(params = null) {
		//this.rid = 0;
		//this.id = null;
		this.readtime = null;
		this.level = Number.NaN;
		this.discharge = Number.NaN;
		//this.prov = null;
		if(params) {
			for(let [k, v] of Object.entries(params))
				this[k] = v;
		}	
	}

	toString() {
		return `[Reading: ${this.id}, ${this.readtime}, ${this.level}, ${this.discharge}]`;
	}
	
	getRid() {
		return rid;
	}
	setRid(rid) {
		this.rid = rid;
	}
	getId() {
		return this.id;
	}
	setId(id) {
		this.id = id;
	}
	getReadTime() {
		return this.readtime;
	}
	setReadTime(readingTime) {
		this.readtime = readingTime;
	}
	getLevel() {
		return this.level;
	}
	setLevel(level) {
		this.level = level;
	}
	getDischarge() {
		return this.discharge;
	}
	setDischarge(discharge) {
		this.discharge = discharge;
	}
	getProvince() {
		return this.prov;
	}
	setProvince(province) {
		this.prov = province;
	}
	
}


module.exports = Reading;
